### DISCUSSION TITLE
Can it be integrated into other open source 3d frameworks, such as osg?

### DISCUSSION THREAD

**Discussion by luohuiiqng** [openframeworks/openFrameworks#8086] [Q&A]
Thank you!

**Comment by artificiel** [comment]
hello! there is a very old addon here https://github.com/stmh/ofxOpenSceneGraph not sure if it's still relevant with current OF/OSG and/or if it fits your expectations.

GitHub discussions don't get a lot of visibility, if you have other questions I recommend the forum https://forum.openframeworks.cc

**Comment by GitBruno** [comment]
To answer your question, generally yes. **openFrameworks is distributed under the MIT License. This gives everyone the freedoms to use openFrameworks in any context: commercial or non-commercial, public or private, open or closed source.** > See [licence](https://github.com/openframeworks/openFrameworks?tab=License-1-ov-file)

However, pending on your platform and usage, you will want to [check all licences](https://github.com/openframeworks/openFrameworks/blob/master/docs/libraries.md).

Thanks @artificiel I didn't know about this plugin. It does interest me, however it makes more sense to implement [OSD](https://openusd.org/release/index.html) over [OSG](https://openscenegraph.github.io/openscenegraph.io/)
